CSharp/Library/Microsoft.Bot.Connector.Shared/MicrosoftAppCredentials.cs

```diff
@@ -168,19 +168,44 @@ public static bool IsTrustedServiceUrl(string serviceUrl)
 
         public async Task<string> GetTokenAsync(bool forceRefresh = false)
         {
-            string token;
             OAuthResponse oAuthToken;
-            if (cache.TryGetValue(TokenCacheKey, out oAuthToken) && !forceRefresh && TokenNotExpired(oAuthToken))
+            bool tokenInCache = cache.TryGetValue(TokenCacheKey, out oAuthToken);
+            string token = tokenInCache ? oAuthToken.access_token : null;
+            if (!tokenInCache || !TokenNotExpired(oAuthToken)) 
             {
-                token = oAuthToken.access_token;
+                try
+                {

```

Comment:
don't need this try{} catch{} block anymore since RereshAndStoreToken(...) has the try catch